src/geo/transform.js

```diff
@@ -1227,8 +1228,7 @@ class Transform {
      */
     pointCoordinate3D(p: Point): MercatorCoordinate {
         if (!this.elevation) return this.pointCoordinate(p);
```

Comment:
Globe.pointCoordinate seem to already clamp to the closest point on the sphere. Is this just preparation for after Terrain.pointCoordinate gets implemented for globe projection? And currently, iterative approach is not needed for globe?

Reply:
I think the original idea for the clamping behavior was to keep the function signature intact. There's no option for returning a null value if nothing was picked. This was also the easiest way forward as many parts of the codebase (such as feature querying) relies on this kind of behavior.

But this is a really good question to ask! Both the function `projection.pointCoordinate` and this `transform.pointCoordinate3D` should be revisited at some point.